src/ng/directive/input.js

```diff
@@ -1180,6 +1186,26 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
   // or form autocomplete on newer browser, we need "change" event to catch it
   element.on('change', listener);
 
+  // Some native input types (date-family) have the ability to change validity without
+  // firing any input/change events.
+  // For these event types, when native validators are present and the browser supports the type,
+  // check for validity changes on various DOM events.
+  if (PARTIAL_VALIDATION_TYPES[type] && ctrl.$$hasNativeValidators && type === attr.type) {
+    element.on(PARTIAL_VALIDATION_EVENTS, function(ev) {
+      if (!timeout) {
```

Comment:
I've had that concern as well for a while now with `keydown` vs `cut`/`paste` events in the IE case (no `input` event).

But in this case, with current browser date-like input support, I don't think there will be any issues. I don't think there is any browser that will *not* have the `input` event but *will* have native date-like inputs.

I do think this should change at some point though, even if just for the IE case...